examples/levels/src/reverse.ts

```diff
@@ -0,0 +1,238 @@
+import { 
+    Connection, 
+    Keypair, 
+    Commitment, 
+    sendAndConfirmTransaction, 
+    Transaction, 
+    PublicKey
+} from "@solana/web3.js";
+import * as dotenv from "dotenv";
+import bs58 from "bs58";
+import { 
+    INITIAL_FDV,
+    INITIAL_QTY_RATIO, 
+    INITIAL_QTY_REDUCTION,
+    QTY_DECR_REDUCTION,
+    INITIAL_PRICE_INCR,
+    PRICE_INCR_REDUCTION,
+    TOKEN_SUPPLY,
+    NUM_LEVELS,
+    MAX_NUM_ORDER_IX
+} from './constants';
+
+dotenv.config();
+
+// Envs
+const PRIVATE_KEY = process.env.PRIVATE_KEY;
+const RPC_URL = process.env.RPC_URL;
+const MFX_MKT = process.env.MFX_MKT;
+
+interface BondingLevel {
+    price: number;
+    quantity: number;
+    level: number;
+}
+
+/**
+ * Generates order levels similar to a bonding curve from AMMs 
+ * Returns the bonding levels corresponding prices and quantities.
+ * Price increases and quantity decreases are reduced at each level.
+ */
+function generateBondingLevels(
+    totalSupply: number = 1_000_000_000,
+    numLevels: number = 100,
+): BondingLevel[] {
+    const levels: BondingLevel[] = [];
+    const initialPrice = INITIAL_FDV / totalSupply;
+    
+    let currentQuantity = totalSupply * INITIAL_QTY_RATIO;
+    let currentPrice = initialPrice;
+    
+    for (let level = 0; level < numLevels; level++) {
+        levels.push({
+            price: currentPrice,
+            quantity: currentQuantity,
+            level: level + 1,
+        });
+        const currentPriceIncrease = Math.max(INITIAL_PRICE_INCR - (level * PRICE_INCR_REDUCTION), 0);
+        const currentQuantityReduction = Math.min(0, INITIAL_QTY_REDUCTION - (level * QTY_DECR_REDUCTION));
+        
+        currentQuantity *= (1 + currentQuantityReduction);
+        currentPrice *= (1 + currentPriceIncrease);
+    }
+
+    console.log('Price Range:', {
+        initialPrice,
+        finalPrice: levels[levels.length - 1].price,
+        priceMultiple: levels[levels.length - 1].price / initialPrice,
+        numLevels: levels.length
+    });
+    
+    return levels;
+}
+
+/**
+ * Gets the current level based on total tokens sold.
+ * Levels are 1-based to match the order ID convention.
+ */
+function getCurrentLevel(
+    totalTokensSold: number,
+    levels: BondingLevel[]
+): number {
+    let accumulatedTokens = 0;
+    
+    for (let i = 0; i < levels.length; i++) {
+        accumulatedTokens += levels[i].quantity;
+        if (accumulatedTokens >= totalTokensSold) {
+            return i + 1;
+        }
+    }
+    
+    return levels.length;
+}
+
+/**
+ * Generates initial orders for the bonding curve using reverse orders.
+ * Each order will automatically flip between bid/ask after being filled.
+ */
+function generateInitialOrders(
+    levels: BondingLevel[],
+    currentLevel: number
+): WrapperPlaceOrderReverseParamsExternal[] {
+    const orders: WrapperPlaceOrderReverseParamsExternal[] = [];
+    
+    // Place bid orders for cleared levels
```

Comment:
Why do we care about cleared levels when this is just initial? is the idea that this is happening for tokens that already have been launched?

Reply:
If for whatever reason we need to switch markets this could be useful. Didn't see a good reason to remove it effectively

Reply:
Why is it useful for switching markets? Isnt the assumption with reverse orders being used that there is no maintenance and we just place all levels, not just up through cleared level.
Remove? This is new code.

Reply:
It could, in our narrow case no, but if a market was write locked or something and forced us to we would want to have this. Low probability edge case. Can remove don't care